digitransit-search-util/packages/digitransit-search-util-query-utils/index.js

```diff
@@ -474,14 +478,52 @@ export function getRoutesQuery(input, feedIds, transportMode, pathOpts) {
     feeds: Array.isArray(feedIds) && feedIds.length > 0 ? feedIds : null,
     name: input,
     modes: transportMode ? modes : null,
-  })
-    .then(data =>
-      data.viewer.routes
-        .map(r => mapRoute(r, pathOpts))
-        .filter(route => !!route)
-        .sort((x, y) => routeNameCompare(x.properties, y.properties)),
-    )
-    .then(suggestions => take(suggestions, 100));
+  }).then(data => {
+    const results = data.viewer.routes
+      .map(r => mapRoute(r, pathOpts))
+      .filter(route => !!route);
+    const normalizedTerm = !isString(input) ? '' : input.toLowerCase();
+    const orderedResults = orderBy(
+      results,
+      [
+        result => {
+          const { confidence, layer, source } = result.properties;
+          if (normalizedTerm.length === 0) {
+            // Search with an empty string
+            return getLayerRank(layer, source);
+          }
+
+          if (!confidence) {
```

Comment:
this query is for OTP, not geocoder, so confidence is always undefined. All logic after this block can be deleted.